Enhanced context format

hungthai1401 opened the pull request:
I have separated each context type to a processor class for scalable such as

```mermaid
graph TD
    A[RpcLogger] --> B[ContextProcessorManager]
    B --> C[NullProcessor]
    B --> D[ScalarProcessor]
    B --> E[DateTimeProcessor]
    B --> F[ThrowableProcessor]
    B --> G[StringableProcessor]
    B --> H[ArrayProcessor]
    B --> I[ResourceProcessor]
    B --> J[ObjectProcessor]
    B --> K[FallbackProcessor]
    B --> L[CustomProcessor]
    
    M[Context Data] --> B
    B --> N[Processed Context]
    N --> O[RoadRunner Logger]
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Context normalization at log time: dates, exceptions, stringables, resources, nested arrays/objects are auto-normalized for consistent structured logs.
  - Pluggable context processor with a sensible default chain and support for custom processor ordering.

- Tests
  - Broadened unit coverage for many context types, processor ordering, and custom processor integration.
  - Test isolation improved with cleanup between tests.

- Chores
  - Test script updated to enable coverage when running tests.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hungthai1401 committed:
feat: implement context processors for structured logging

- Added ContextProcessorManager to manage multiple context processors.
- Introduced various processors: ArrayProcessor, DateTimeProcessor, FallbackProcessor, NullProcessor, ObjectProcessor, ResourceProcessor, ScalarProcessor, StringableProcessor, and ThrowableProcessor.
- Enhanced RpcLogger to utilize ContextProcessorManager for processing log context data.

hungthai1401 committed:
test: add comprehensive tests for context processors

- Implemented unit tests for various context processors including ArrayProcessor, DateTimeProcessor, FallbackProcessor, NullProcessor, ObjectProcessor, ResourceProcessor, ScalarProcessor, StringableProcessor, and ThrowableProcessor.
- Ensured coverage for processing different data types and edge cases, validating the functionality of each processor.
- Enhanced the RpcLoggerTest with additional tests for logging with complex contexts and custom processors.

roxblnfk reviewed (commented):
Hello. I'm thinking about how we can optimize this. The approach with processors is quite flexible, but we need a very high-performance logger because I plan to use it in Temporal Workflows. I think we should simplify some things for better performance.

1. We should move the processors to the Internal folder and mark all classes as `@internal`. This way, we can redesign them in the future without breaking backward compatibility.
2. I think we should handle built-in types (like null, string, int, float, bool, array) in one processor or even directly in the manager. We need processors only for objects of specific classes, right?

hungthai1401 commented:
@roxblnfk
1. Ok, I got it.
2. I have ScalarProcessor for built-in types, we can merge NullProcess to it, but I think array and object need recursive processors and we should separate them

Anw, what do you think if we allow adding specific processors in user view?

roxblnfk commented:
> array needs recursive processor

That's why builtin types might be processed right in the manager.
And processors can now be focused only on objects

hungthai1401 committed:
feat: implement context processors for structured logging with array, scalar, and resource handling

hungthai1401 committed:
refactor: consolidate array, null, and scalar processing into BuiltInTypeProcessor

- Removed ArrayProcessor, NullProcessor, and ScalarProcessor.
- Introduced BuiltInTypeProcessor to handle null, scalar, and array types efficiently.
- Updated ContextProcessorManager to register BuiltInTypeProcessor.
- Added unit tests for BuiltInTypeProcessor to ensure correct functionality.

hungthai1401 commented:
@roxblnfk I have updated them, pls confirm again. tysm

roxblnfk commented on composer.json:
```diff
@@ -43,7 +43,7 @@
         "cs:diff": "php-cs-fixer fix --dry-run -v --diff --show-progress dots",
         "cs:fix": "php-cs-fixer fix -v",
-        "test": "phpunit",
+        "test": "XDEBUG_MODE=coverage phpunit",
```
```suggestion
        "test": [
            "@putenv XDEBUG_MODE=coverage",
            "phpunit"
        ],
```

hungthai1401 commented:
@roxblnfk do you have any updates?

roxblnfk committed:
refactor: simplify context processors

roxblnfk committed:
test: Update tests

roxblnfk committed:
feat: Make processors non-internal

roxblnfk committed:
chore: Fix `test` script in `composer.json`

roxblnfk committed:
chore: Cover DefaultProcessor with comments

roxblnfk committed:
chore: Add symfony/polyfill-php83 for the `Override` attribute
